From f8d78339a31c71bcfa49385506eb016d75e06dec Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 10 Mar 2020 10:37:12 -0700
Subject: [PATCH 1268/1852] v4l: xilinx: dma: Prevent divide by zero error

commit c201032fcf163db6dd76472dce23ea54ed5e36ea from
https://github.com/Xilinx/linux-xlnx.git

Currently there is no issue, but there is a possibility of a divide by
zero. This was found during a coverity check which gave the
following error:

"Event zero_return: Function call "lcm(dma->align, info->bpp >> 3)" returns 0."

This patch fixes this by adding a check to see if the calculated divisor
is 0.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index cfdc1034d2e0..441e3d56f499 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -1000,6 +1000,13 @@ __xvip_dma_try_format(struct xvip_dma *dma,
 	 * it back to pixels.
 	 */
 	align = lcm(dma->align, info->bpp >> 3);
+	if (!align) {
+		dev_err(dma->xdev->dev,
+			"transfer alignment is 0: dma->align = %x, bpp = %u\n",
+			dma->align, info->bpp);
+		return;
+	}
+
 	min_width = roundup(XVIP_DMA_MIN_WIDTH, align);
 	max_width = rounddown(XVIP_DMA_MAX_WIDTH, align);
 
-- 
2.31.1

